cmd/dfget/app/root.go

```diff
@@ -0,0 +1,207 @@
+	if cfg.Ctx.Console {
+		util.AddConsoleLog(cfg.Ctx.ClientLogger)
+	}
+	if cfg.Ctx.Pattern == "p2p" {
+		cfg.Ctx.ServerLogger = util.CreateLogger(logPath, "dfserver.log", logLevel, cfg.Ctx.Sign)
+	}
+}
+
+func initFlags() {
+	// url & output
+	rootCmd.PersistentFlags().StringVarP(&cfg.Ctx.URL, "url", "u", "",
+		"will download a file from this url")
+	rootCmd.PersistentFlags().StringVarP(&cfg.Ctx.Output, "output", "o", "",
+		"output path that not only contains the dir part but also name part")
+
+	// localLimit & totalLimit & timeout
+	localLimit := rootCmd.PersistentFlags().StringP("locallimit", "s", "",
+		"rate limit about a single download task, its format is 20M/m/K/k")
+	totalLimit := rootCmd.PersistentFlags().String("totallimit", "",
+		"rate limit about the whole host, its format is 20M/m/K/k")
+	rootCmd.PersistentFlags().IntVarP(&cfg.Ctx.Timeout, "timeout", "e", 0,
+		"download timeout(second)")
+
+	// md5 & identifier
+	rootCmd.PersistentFlags().StringVarP(&cfg.Ctx.Md5, "md5", "m", "",
+		"expected file md5")
+	rootCmd.PersistentFlags().StringVarP(&cfg.Ctx.Identifier, "identifier", "i", "",
+		"identify download task, it is available merely when md5 param not exist")
+
+	rootCmd.PersistentFlags().StringVar(&cfg.Ctx.CallSystem, "callsystem", "",
+		"system name that executes dfget")
+
+	rootCmd.PersistentFlags().StringVarP(&cfg.Ctx.Pattern, "pattern", "p", "p2p",
+		"download pattern, must be 'p2p' or 'cdn' or 'source'"+
+			"\ncdn/source pattern not support 'totallimit' flag")
+
+	filter := rootCmd.PersistentFlags().StringP("filter", "f", "",
+		"filter some query params of url, use char '&' to separate different params"+
+			"\neg: -f 'key&sign' will filter 'key' and 'sign' query param"+
+			"\nin this way, different urls correspond one same download task that can use p2p mode")
+
+	rootCmd.PersistentFlags().StringSliceVar(&cfg.Ctx.Header, "header", nil,
+		"http header, eg: --header='Accept: *' --header='Host: abc'")
+
+	rootCmd.PersistentFlags().StringSliceVarP(&cfg.Ctx.Node, "node", "n", nil,
+		"specify supnernodes")
+
+	rootCmd.PersistentFlags().BoolVar(&cfg.Ctx.Notbs, "notbs", false,
+		"not back source when p2p fail")
+	rootCmd.PersistentFlags().BoolVar(&cfg.Ctx.DFDaemon, "dfdaemon", false,
+		"caller is from dfdaemon")
+
+	// others
+	rootCmd.PersistentFlags().BoolVarP(&cfg.Ctx.Version, "version", "v", false,
+		"show version")
+	rootCmd.PersistentFlags().BoolVarP(&cfg.Ctx.ShowBar, "showbar", "b", false,
+		"show progress bar, it's conflict with '--console'")
+	rootCmd.PersistentFlags().BoolVar(&cfg.Ctx.Console, "console", false,
+		"show log on console, it's conflict with '--showbar'")
+	rootCmd.PersistentFlags().BoolVar(&cfg.Ctx.Verbose, "verbose", false,
+		"be verbose")
+	rootCmd.PersistentFlags().BoolVarP(&cfg.Ctx.Help, "help", "h", false,
+		"show help information")
+
+	rootCmd.PersistentFlags().MarkDeprecated("exceed", "please use '--timeout' or '-e' instead")
+	// be compatible with dfget python version
```

Comment:
Could please help get it out of the function of `initFlags`, It's wired here. ^_^

Reply:
Remove or Put it into an other place?